logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/notificationToken/ObserveNotificationTokenUpdating.kt

```diff
@@ -0,0 +1,46 @@
+package com.wire.kalium.logic.feature.notificationToken
+
+import com.wire.kalium.logic.configuration.notification.NotificationTokenRepository
+import com.wire.kalium.logic.data.client.ClientRepository
+import com.wire.kalium.logic.data.user.UserRepository
+import com.wire.kalium.logic.functional.flatMap
+import com.wire.kalium.logic.functional.isRight
+import com.wire.kalium.network.api.user.pushToken.PushTokenBody
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+
+internal interface ObserveNotificationTokenUpdating {
```

Comment:
As this is not a UseCase, we can name it like a normal type and not like a function.

```suggestion
internal interface PushTokenUpdater
```